File: mobile/src/main/java/com/backdoor/moove/core/helper/Module.java
package com.backdoor.moove.core.helper;

import android.os.Build;

/**
 * Helper class for checking type of built application.
 */
public class Module {

    /**
     * Check if device runs on JellyBean 2 and above.
     *
     * @return boolean
     */
    public static boolean isJellyBean() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2;
    }

    /**
     * Check if device runs on Lollipop and above.
     *
     * @return boolean
     */
    public static boolean isLollipop() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
    }

    /**
     * Check if device runs on Marshmallow and above.
     *
     * @return boolean
     */
    public static boolean isMarshmallow() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M;
    }

    /**
     * Check if device runs on Lollipop and above.
     *
     * @return boolean
     */
    public static boolean isKitkat() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT;
    }

    public static boolean isNougat() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.N;
    }

    public static boolean isO() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.O;
    }
}

File: mobile/src/main/java/com/backdoor/moove/core/helper/Module.kt
package com.backdoor.moove.core.helper

import android.os.Build

/**
 * Helper class for checking type of built application.
 */
object Module {

    /**
     * Check if device runs on JellyBean 2 and above.
     *
     * @return boolean
     */
    val isJellyBean: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2

    /**
     * Check if device runs on Lollipop and above.
     *
     * @return boolean
     */
    val isLollipop: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP

    /**
     * Check if device runs on Marshmallow and above.
     *
     * @return boolean
     */
    val isMarshmallow: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.M

    /**
     * Check if device runs on Lollipop and above.
     *
     * @return boolean
     */
    val isKitkat: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT

    val isNougat: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.N

    val isO: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.O
}
